src/Bicep.Core.Samples/Files/InvalidModules_LF/Completions/moduleATopLevelProperties.json

```diff
@@ -5,7 +5,7 @@
     "detail": "dependsOn",
     "documentation": {
       "kind": "markdown",
-      "value": "Type: `resource | module[]`  \nWrite-only property  \n"
+      "value": "Type: `(module[] | resource | module | resource[])[]`  \nWrite-only property  \n"
```

Comment:
The ordering is a little weird - presumably this is what's shown to the user. Can we order this somehow?

Reply:
Should be able to. I assume you're looking for `resource | resource[] | module | module[]`?

---
In reply to: [582066499](https://github.com/Azure/bicep/pull/1521#discussion_r582066499) [](ancestors = 582066499)

Reply:
Either that or `resource | module | resource[] | module[]` - both seem like good options.

Reply:
It's actually more complicated than I thought because `resource | module` part is one member of the union. (Due to how ResourceRef type is implemented.) We can improve this later.

---
In reply to: [582234791](https://github.com/Azure/bicep/pull/1521#discussion_r582234791) [](ancestors = 582234791)

Reply:
> It's actually more complicated than I thought because `resource | module` part is one member of the union. (Due to how ResourceRef type is implemented.) We can improve this later.
> 
> In reply to: [582234791](https://github.com/Azure/bicep/pull/1521#discussion_r582234791) [](ancestors = 582234791)

So it's really `((resource | module) | resource[] | module[])` flattened into one union?

Reply:
Yeah unfortunately. I really wish we didn't do it that way now :(

---
In reply to: [582331000](https://github.com/Azure/bicep/pull/1521#discussion_r582331000) [](ancestors = 582331000)

Reply:
I think I'll add a special case to enclose it in parentheses for now.

---
In reply to: [582332560](https://github.com/Azure/bicep/pull/1521#discussion_r582332560) [](ancestors = 582332560,582331000)